misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt

```diff
@@ -106,6 +108,40 @@ class HibernateModule(
       )
     }).asSingleton()
 
+    if (readerQualifier != null) {
+      val readerTransacterKey = Transacter::class.toKey(readerQualifier)
+      bind(readerTransacterKey).toProvider(object : Provider<Transacter> {
+        @Inject lateinit var executorServiceFactory: ExecutorServiceFactory
+        override fun get(): Transacter = RealTransacter(
+            qualifier = readerQualifier,
+            sessionFactoryProvider = readerSessionFactoryProvider!!,
+            readerSessionFactoryProvider = readerSessionFactoryProvider,
+            config = config,
+            executorServiceFactory = object : ExecutorServiceFactory {
```

Comment:
I wonder if we need this abstraction somewhere else.

Reply:
Let's bind this transacter only for MySQL. TiDB and Vitess replicas both work differently and this is not the correct configuration. That will also avoid your problem with failing Vitess tests

Reply:
If we do this I'm going to need to skip the Reader Transacter test for vitess somehow.